p2p/peer.py

```diff
@@ -396,7 +396,7 @@ def decrypt_header(self, data: bytes) -> bytes:
         self.ingress_mac.update(sxor(aes, header_ciphertext))
         expected_header_mac = self.ingress_mac.digest()[:HEADER_LEN]
         if not bytes_eq(expected_header_mac, header_mac):
-            raise AuthenticationError('Invalid header mac')
+            raise DecryptionError('Invalid header mac')
```

Comment:
Curious why the exception class here was changed?

Reply:
https://github.com/ethereum/py-evm/pull/602#discussion_r184365395

Both seemed to mean the same thing so I got rid of one.